src/pages/index.js

```diff
@@ -56,28 +74,121 @@ const enableValidation = (config) => {
   });
 };
 enableValidation(config);
-
 /**============================================
  *               Functions
  *=============================================**/
+function handleApiCardInfo(cardData) {
+  const cardApiSection = new Section(
+    { items: cardData, renderer: renderCard },
+    cardContainer
+  );
+  cardApiSection.renderItems();
+}
 function renderCard(item, method = "append") {
-  const card = new Card(item, cardSelector, handleImageClick);
+  const card = new Card(
+    item,
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardElement = card.renderCard();
   cardSection.addItem(cardElement, method);
 }
 function handleImageClick(data) {
   popupWithImage.open(data);
 }
-function handleProfileSubmit(userData) {
+function handleDeleteClick(cardData) {
+  deleteConfirmation.open(cardData);
+}
+function deleteConfirmed(card) {
+  api
+    .deleteCard(card.cardID)
+    .then(() => {
+      card.cardElement.remove();
+      card.cardElement = null;
+    })
+    .then(deleteConfirmation.close())
+    .catch(console.error);
+}
+function handleLikeClick(cardData, card, likeAction) {
+  if (likeAction === "addLike") {
+    api
+      .addLike(card._id)
+      .then(cardData.likeButton.classList.add("card__like-button_active"))
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+  if (likeAction === "removeLike") {
+    api
+      .deleteLike(card._id)
+      .then(cardData.likeButton.classList.remove("card__like-button_active"))
```

Comment:
Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle

Reply:
Ahh I see, I have moved this code into the Card.js file and corrected the logic.